### ISSUE TITLE
Inconsistent image size with savefig(...,type='png')

### ISSUE THREAD

**Issue by sorenrasmussen** [matplotlib/matplotlib#8542]
### Bug report

**Bug summary**

`savefig()` with `type='png'` produces different image sizes, depending on the `dpi` argument when calling `figure()`.

**Code for reproduction**
```python
from pylab import *
%matplotlib tk

plt.figure(figsize=(1,1),dpi=100)
plt.savefig("100-300dpi.png",type="png",dpi=300)

plt.figure(figsize=(1,1),dpi=96) #96 is my screen DPI
plt.savefig("96-300dpi.png",type="png",dpi=300)
```

**Actual outcome**

100-300dpi.png is 654x300 pixels.
96-300dpi.png is 681x300 pixels.

**Expected outcome**

I would expect both to come out at 300x300 pixels, or at least equal size.

**Matplotlib version**

  * Matplotlib 2.0.0
  * Python 2.7.13
  * IPython 5.1.0
  * Jupyter Version: 4.4.1
  * Platform: Linux

**Comment by sorenrasmussen** [comment]
Perhaps this is a better example, resulting in _almost_ equal widths:

```python
from pylab import *
%matplotlib tk
figsize = (10/3.0, 1)
plt.figure(figsize=figsize,dpi=100)
plt.savefig("100-300dpi.png",type="png",dpi=300)
plt.figure(figsize=figsize,dpi=96)
plt.savefig("96-300dpi.png",type="png",dpi=300)
```
I would expect both to come out at 1000x300 px.

**Comment by jenshnielsen** [comment]
This is causes by using an interactive backend. (TK in this case) The menu bar has a certain minimum width in inches so the figure can't be any smaller than that in that direction so it is dynamically resized to fit the width. Without tkagg it produces exactly what you expect (300X300)

```
matplotlib.use('Agg')
import matplotlib.pyplot as plt

plt.figure(figsize=(1,1),dpi=100)
plt.savefig("100-300dpi.png",type="png",dpi=300)

plt.figure(figsize=(1,1),dpi=96) #96 is my screen DPI
plt.savefig("96-300dpi.png",type="png",dpi=300)

```

**Comment by sorenrasmussen** [comment]
How about the second example in my comment then?
The width is 3.333... inches, which leaves plenty of space for the TK menu bar

**Comment by jenshnielsen** [comment]
I doubt that's related to the first issue. It looks very much like a round off error somewhere in the conversion

**Comment by sorenrasmussen** [comment]
Should I post that as a new issue then, or just edit this one (since the menu bar thing is sort of a non-issue)?

**Comment by jenshnielsen** [comment]
It's probably cleanest to make a new issue but it's not important as long as it's clear what it's about

**Comment by sorenrasmussen** [comment]
Closed: Replaced by [#8543](https://github.com/matplotlib/matplotlib/issues/8543)
